pyo3-macros-backend/src/pymethod.rs

```diff
@@ -609,8 +609,21 @@ pub fn impl_py_setter_def(
         })
     })
     .unwrap_or_else(|| {
+        let (span, name) = match &property_type {
+            PropertyType::Descriptor { field, .. } => (field.ty.span(), field.ident.as_ref().map(|i|i.to_string()).unwrap_or_default()),
+            PropertyType::Function { spec, .. } => {
+                let (_, args) = split_off_python_arg(&spec.signature.arguments);
+                (args[0].ty.span(), args[0].name.to_string())
+            }
+        };
+
+        let holder = holders.push_holder(span);
```

Comment:
Maybe add a TODO to rework this to use `impl_arg_param` later? I think the hardest part to doing so would be that `impl_arg_param` assumes the arguments are in an array and takes the array ident plus an index. We could probably rework that so that `impl_arg_param` just gets a `TokenStream` of something to extract from.

(Of course, if you're up for it, can also go for the refactoring now 👀)